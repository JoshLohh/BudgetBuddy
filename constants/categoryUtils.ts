import { Ionicons } from "@expo/vector-icons";

export const CATEGORIES = [
  { label: 'Food', value: 'Food', icon: 'restaurant' },
  { label: 'Transportation', value: 'Transportation', icon: 'car' },
  { label: 'Accommodation', value: 'Accommodation', icon: 'home' },
  { label: 'Entertainment', value: 'Entertainment', icon: 'film' },
  { label: 'Utilities', value: 'Utilities', icon: 'bulb' },
  { label: 'Shopping', value: 'Shopping', icon: 'cart' },
  { label: 'Others', value: 'Others', icon: 'ellipsis-horizontal' },
];

export const getCategoryIconName = (category: string) => {
  const found = CATEGORIES.find(c => c.value === category);
  return found?.icon || 'ellipsis-horizontal';
};

export type IoniconName = React.ComponentProps<typeof Ionicons>['name'];