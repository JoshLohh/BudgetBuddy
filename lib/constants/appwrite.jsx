import { Client, Account, ID, Avatars } from 'react-native-appwrite';


export const client = new Client
    .setProject("6838ad0d003d2a0754a0")
    .setPlatform('dev.Buddies.budgetbuddy');

export const account = new Account(client)
export const avatars = new Avatars(client)