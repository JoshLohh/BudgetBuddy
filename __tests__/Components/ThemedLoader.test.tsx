import React from 'react';
import { render } from '@testing-library/react-native';
import * as ReactNative from 'react-native';
import ThemedLoader from '../../components/ThemedLoader';
import { Colors } from '../../constants/Colors';
import { View } from 'react-native';

// Fixed: always assign the test ID to the mock ThemedView
jest.mock('../../components/ThemedView', () => {
    const React = require('react'); // Import React here
    const { View } = require('react-native'); // Import View here, inside the factory
    return {
      ThemedView: ({ children, testID = 'themed-view', ...props }: { children?: React.ReactNode; testID?: string }) => (
        <View testID={testID} {...props}>
          {children}
        </View>
      ),
    };
  });

describe('ThemedLoader', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('renders with light theme and uses light text color for ActivityIndicator', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('light');
    const { getByTestId } = render(<ThemedLoader />);
    const indicator = getByTestId('activity-indicator');
    expect(indicator.props.color).toBe(Colors.light.text);
  });

  it('renders with dark theme and uses dark text color for ActivityIndicator', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('dark');
    const { getByTestId } = render(<ThemedLoader />);
    const indicator = getByTestId('activity-indicator');
    expect(indicator.props.color).toBe(Colors.dark.text);
  });

  it('falls back to light theme when color scheme is undefined', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue(undefined);
    const { getByTestId } = render(<ThemedLoader />);
    const indicator = getByTestId('activity-indicator');
    expect(indicator.props.color).toBe(Colors.light.text);
  });

  it('renders ThemedView as parent', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('light');
    const { getByTestId } = render(<ThemedLoader />);
    const themedView = getByTestId('themed-view');
    expect(themedView).toBeTruthy();
  });

  it('matches the snapshot for light theme', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('light');
    const tree = render(<ThemedLoader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches the snapshot for dark theme', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('dark');
    const tree = render(<ThemedLoader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

/**
 * This test file asserts that ThemedLoader renders correctly against the theme logic,
 * and verifies ThemedView is always present as the parent container.
 */
