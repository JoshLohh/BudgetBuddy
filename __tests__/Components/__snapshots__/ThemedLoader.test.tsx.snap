// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedLoader matches the snapshot for dark theme 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="themed-view"
>
  <ActivityIndicator
    color="#ECEDEE"
    size="large"
    testID="activity-indicator"
  />
</View>
`;

exports[`ThemedLoader matches the snapshot for light theme 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="themed-view"
>
  <ActivityIndicator
    color="#11181C"
    size="large"
    testID="activity-indicator"
  />
</View>
`;
